### PR TITLE
[gardening] Remove unused variables

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

**Comment by practicalswift** [comment]
@swift-ci please test

File: lib/IDE/SwiftSourceDocInfo.cpp [hunk 1]
```diff
@@ -290,13 +290,13 @@ static bool hasUnhandledError(ArrayRef<ASTNode> Nodes) {
         if (DCS->isSyntacticallyExhaustive())
           return false;
         Throwing = true;
-      } else if (auto TS = dyn_cast<ThrowStmt>(S)) {
+      } else if (dyn_cast<ThrowStmt>(S)) {
```
Comment (slavapestov) [inline lib/IDE/SwiftSourceDocInfo.cpp]: If the result is unused, please use isa instead of dyn_cast

**Comment by practicalswift** [comment]
@slavapestov Good point, fixed!

**Comment by practicalswift** [comment]
@swift-ci please smoke test and merge

**Comment by practicalswift** [comment]
@swift-ci please test

**Pull request merged**